### PR TITLE
refactor: move ipc request validation and authZ to ipc class

### PR DESCRIPTION
**Pull request by saranyailla**
**Issue #, if available:**

**Description of changes:**
Move IPC request validation, authZ and handling to IPC agent class. 

**Why is this change necessary:**

**How was this change tested:**

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java [hunk 1]
```diff
@@ -5,32 +5,124 @@
...
+                    OBJECT_MAPPER.readValue(request,
+                            com.aws.greengrass.secretmanager.model.v1.GetSecretValueRequest.class);
+            validateSecretIdAndDoAuthorization(GET_SECRET_VALUE, serviceName, getSecretValueRequest.getSecretId());
+            GetSecretValueResult response = secretManager.getSecret(getSecretValueRequest);
+
+            try {
+                return OBJECT_MAPPER.writeValueAsBytes(response);
+            } catch (JsonProcessingException e) {
+                // not logging exception in case it outputs secret value
+                logger.atError().event("secret-access").kv("Principal", serviceName).kv("secret", new String(request))
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java:84]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

#### new String [](|8df4fb67-a24a-4a26-a18a-395ef71250d0|)
Java Strings use Unicode internally to encode characters. 

When constructing a string from a `byte[]`, a character encoding must be used to convert the bytes to a String&#39;s internal encoding. When no character encoding is passed explicitly, the Java runtime will use a default encoding. This encoding usually isn&#39;t UTF-8 and might become misconfigured. When converting bytes to characters or strings, always supply the actual character encoding to use.

#### Suggested solutions:
Specify an encoding

*(likely `UTF-8`, which is backward compatible with ASCII and has multi-language support)*

```
String message = new String(messageBytes, StandardCharsets.UTF_8);
```

File: src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java [hunk 2]
```diff
@@ -5,32 +5,124 @@
...
+            logger.atError().event("secret-access").setCause(t).kv("Principal", serviceName)
+                    .kv("secret", new String(request)).log("Error getting secret");
+        }
+
+        try {
+            return OBJECT_MAPPER.writeValueAsBytes(
+                    GetSecretValueError.builder().status(status).message(message).build());
+        } catch (JsonProcessingException e) {
+            logger.atError().event("secret-access").setCause(e).kv("Principal", serviceName)
+                    .kv("secret", new String(request)).kv("status", status).kv("message", message)
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java:109]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

#### new String [](|8df4fb67-a24a-4a26-a18a-395ef71250d0|)
Java Strings use Unicode internally to encode characters. 

When constructing a string from a `byte[]`, a character encoding must be used to convert the bytes to a String&#39;s internal encoding. When no character encoding is passed explicitly, the Java runtime will use a default encoding. This encoding usually isn&#39;t UTF-8 and might become misconfigured. When converting bytes to characters or strings, always supply the actual character encoding to use.

#### Suggested solutions:
Specify an encoding

*(likely `UTF-8`, which is backward compatible with ASCII and has multi-language support)*

```
String message = new String(messageBytes, StandardCharsets.UTF_8);
```

File: src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java [hunk 3]
```diff
@@ -5,32 +5,124 @@
...
+            status = 403;
+            message = t.getMessage();
+        } catch (GetSecretException t) {
+            status = t.getStatus();
+            message = t.getMessage();
+        } catch (Throwable t) {
+            status = 500;
+            message = t.getMessage();
+            logger.atError().event("secret-access").setCause(t).kv("Principal", serviceName)
+                    .kv("secret", new String(request)).log("Error getting secret");
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java:101]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

#### new String [](|8df4fb67-a24a-4a26-a18a-395ef71250d0|)
Java Strings use Unicode internally to encode characters. 

When constructing a string from a `byte[]`, a character encoding must be used to convert the bytes to a String&#39;s internal encoding. When no character encoding is passed explicitly, the Java runtime will use a default encoding. This encoding usually isn&#39;t UTF-8 and might become misconfigured. When converting bytes to characters or strings, always supply the actual character encoding to use.

#### Suggested solutions:
Specify an encoding

*(likely `UTF-8`, which is backward compatible with ASCII and has multi-language support)*

```
String message = new String(messageBytes, StandardCharsets.UTF_8);
```

File: src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java [hunk 4]
```diff
@@ -5,32 +5,124 @@
...
 
+    /**
+     * Handles secret API calls from lambda-manager for v1 style lambdas.
+     *
+     * @param serviceName Lambda component name requesting the secret.
+     * @param request     v1 style get secret request
+     * @return v1 style secret or error serialized as JSON bytes. If a serialization error occurs, null is returned.
+     */
+    public byte[] getSecret(String serviceName, byte[] request) {
+        logger.atInfo().event("secret-access").kv("Principal", serviceName).kv("secret", new String(request))
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java:67]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

#### new String [](|8df4fb67-a24a-4a26-a18a-395ef71250d0|)
Java Strings use Unicode internally to encode characters. 

When constructing a string from a `byte[]`, a character encoding must be used to convert the bytes to a String&#39;s internal encoding. When no character encoding is passed explicitly, the Java runtime will use a default encoding. This encoding usually isn&#39;t UTF-8 and might become misconfigured. When converting bytes to characters or strings, always supply the actual character encoding to use.

#### Suggested solutions:
Specify an encoding

*(likely `UTF-8`, which is backward compatible with ASCII and has multi-language support)*

```
String message = new String(messageBytes, StandardCharsets.UTF_8);
```

File: src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java [hunk 5]
```diff
@@ -5,32 +5,124 @@
...
 import javax.inject.Inject;
 
+import static software.amazon.awssdk.aws.greengrass.GreengrassCoreIPCService.GET_SECRET_VALUE;
+
 /**
  * Class to handle business logic for all SecretManager requests over IPC.
  */
 public class SecretManagerIPCAgent {
     private static final Logger logger = LogManager.getLogger(SecretManagerIPCAgent.class);
+    private static final ObjectMapper OBJECT_MAPPER =
```
Comment (jcosentino11) [inline src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java:38]: ~~can we reuse an object mapper provided by nucleus? i forget where it is off the top of my head.  we should also have object mapper allow unknown properties, which allows backwards compatibility if new fields are added~~

Comment (jcosentino11) [reply src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java:38]: ah nevermind, there's already one in `SecretManagerService`, lets have both places referring to the same mapper then

**Review by jcosentino11 — DISMISSED** [review]

**Comment by MikeDombo** [comment]
This change breaks compatibility with lambda manager. Lambda can call to get secrets using Java reflection, looking at getSecret in the service class.

File: src/main/java/com/aws/greengrass/secretmanager/SecretManagerService.java [hunk 6]
```diff
@@ -233,15 +227,15 @@ public byte[] getSecret(String serviceName, byte[] request) {
...
+                    getSecretValueRequest.getSecretId());
             GetSecretValueResult response = secretManager.getSecret(getSecretValueRequest);
 
             try {
                 return OBJECT_MAPPER.writeValueAsBytes(response);
             } catch (JsonProcessingException e) {
                 // not logging exception in case it outputs secret value
-                logger.atError().event("secret-access")
-                        .kv("Principal", serviceName).kv("secret", new String(request))
+                logger.atError().event("secret-access").kv("Principal", serviceName).kv("secret", new String(request))
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/secretmanager/SecretManagerService.java:238]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

#### new String [](|8df4fb67-a24a-4a26-a18a-395ef71250d0|)
Java Strings use Unicode internally to encode characters. 

When constructing a string from a `byte[]`, a character encoding must be used to convert the bytes to a String&#39;s internal encoding. When no character encoding is passed explicitly, the Java runtime will use a default encoding. This encoding usually isn&#39;t UTF-8 and might become misconfigured. When converting bytes to characters or strings, always supply the actual character encoding to use.

#### Suggested solutions:
Specify an encoding

*(likely `UTF-8`, which is backward compatible with ASCII and has multi-language support)*

```
String message = new String(messageBytes, StandardCharsets.UTF_8);
```

File: src/main/java/com/aws/greengrass/secretmanager/SecretManagerService.java [hunk 7]
```diff
@@ -216,14 +209,15 @@
...
+     * @param request     v1 style get secret request
      * @return v1 style secret or error serialized as JSON bytes. If a serialization error occurs, null is returned.
      */
+    // Note: Do not rename/move this method around as it is used via reflection from the lambda-manager.
     public byte[] getSecret(String serviceName, byte[] request) {
-        logger.atInfo().event("secret-access")
-                .kv("Principal", serviceName)
-                .kv("secret", new String(request))
+
+        logger.atInfo().event("secret-access").kv("Principal", serviceName).kv("secret", new String(request))
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/secretmanager/SecretManagerService.java:220]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

#### new String [](|8df4fb67-a24a-4a26-a18a-395ef71250d0|)
Java Strings use Unicode internally to encode characters. 

When constructing a string from a `byte[]`, a character encoding must be used to convert the bytes to a String&#39;s internal encoding. When no character encoding is passed explicitly, the Java runtime will use a default encoding. This encoding usually isn&#39;t UTF-8 and might become misconfigured. When converting bytes to characters or strings, always supply the actual character encoding to use.

#### Suggested solutions:
Specify an encoding

*(likely `UTF-8`, which is backward compatible with ASCII and has multi-language support)*

```
String message = new String(messageBytes, StandardCharsets.UTF_8);
```

File: src/main/java/com/aws/greengrass/secretmanager/SecretManagerService.java [hunk 8]
```diff
@@ -257,71 +251,18 @@
         } catch (Throwable t) {
             status = 500;
             message = t.getMessage();
-            logger.atError().event("secret-access").setCause(t)
-                    .kv("Principal", serviceName).kv("secret", new String(request))
-                    .log("Error getting secret");
+            logger.atError().event("secret-access").setCause(t).kv("Principal", serviceName)
+                    .kv("secret", new String(request)).log("Error getting secret");
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/secretmanager/SecretManagerService.java:255]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

#### new String [](|8df4fb67-a24a-4a26-a18a-395ef71250d0|)
Java Strings use Unicode internally to encode characters. 

When constructing a string from a `byte[]`, a character encoding must be used to convert the bytes to a String&#39;s internal encoding. When no character encoding is passed explicitly, the Java runtime will use a default encoding. This encoding usually isn&#39;t UTF-8 and might become misconfigured. When converting bytes to characters or strings, always supply the actual character encoding to use.

#### Suggested solutions:
Specify an encoding

*(likely `UTF-8`, which is backward compatible with ASCII and has multi-language support)*

```
String message = new String(messageBytes, StandardCharsets.UTF_8);
```

File: src/main/java/com/aws/greengrass/secretmanager/SecretManagerService.java [hunk 9]
```diff
@@ -257,71 +251,18 @@
...
+            return OBJECT_MAPPER.writeValueAsBytes(
+                    GetSecretValueError.builder().status(status).message(message).build());
         } catch (JsonProcessingException e) {
-            logger.atError().event("secret-access").setCause(e)
-                    .kv("Principal", serviceName)
-                    .kv("secret", new String(request))
-                    .kv("status", status)
-                    .kv("message", message)
+            logger.atError().event("secret-access").setCause(e).kv("Principal", serviceName)
+                    .kv("secret", new String(request)).kv("status", status).kv("message", message)
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/secretmanager/SecretManagerService.java:263]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

#### new String [](|8df4fb67-a24a-4a26-a18a-395ef71250d0|)
Java Strings use Unicode internally to encode characters. 

When constructing a string from a `byte[]`, a character encoding must be used to convert the bytes to a String&#39;s internal encoding. When no character encoding is passed explicitly, the Java runtime will use a default encoding. This encoding usually isn&#39;t UTF-8 and might become misconfigured. When converting bytes to characters or strings, always supply the actual character encoding to use.

#### Suggested solutions:
Specify an encoding

*(likely `UTF-8`, which is backward compatible with ASCII and has multi-language support)*

```
String message = new String(messageBytes, StandardCharsets.UTF_8);
```

File: src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java (hunk 5 above)
Comment (saranyailla) [reply src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java]: Removed it.

File: src/test/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgentTest.java [hunk 10]
```diff
@@ -1,82 +0,0 @@
-/*
```
Comment (MikeDombo) [inline src/test/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgentTest.java:1]: we don't need a test anymore?

Comment (saranyailla) [reply src/test/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgentTest.java:1]: Removed as there's only one test in this file and that is already covered in integ tests with less mocking.

**Review by jcosentino11 — APPROVED** [review]

**Pull request merged**